pandas/core/algorithms.py

```diff
@@ -439,6 +439,10 @@ def unique_with_mask(values, mask: npt.NDArray[np.bool_] | None = None):
         # Dispatch to extension dtype's unique.
         return values.unique()
 
+    if isinstance(values, ABCIndex):
+        # Dispatch to Index's unique.
+        return values.unique()
```

Comment:
Is mask always None if we get here?

Reply:
Yes, I think so. `pd.unique` does not use mask when calling this function.
https://github.com/pandas-dev/pandas/blob/4692686a750b95884e1d5a4f1313614c71c57213/pandas/core/algorithms.py#L408